React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm

```diff
@@ -180,6 +183,30 @@ - (void)ensurePresentedOnlyIfNeeded
   }
 }
 
+- (void)handleDetent:(ModalHostViewDetent)detent
+{
+  if (@available(iOS 15.0, *)) {
+    if (!self.viewController.sheetPresentationController) {
+      return;
+    }
+
+    NSArray<UISheetPresentationControllerDetent *> *detents = @[UISheetPresentationControllerDetent.largeDetent];
+    switch (detent) {
+      case ModalHostViewDetent::Large:
+        break;
+      case ModalHostViewDetent::Medium:
+        detents = @[UISheetPresentationControllerDetent.mediumDetent];
```

Comment:
From @sammy-SC:
> here it looks like we're holding some state in view controller that does not get reset when component is reused or detent value changes. Is that expected?

I think we should update the `prepareForRecycle` method.

Reply:
I tried changing the detent values using the states and they are working as they should. Also, the `prepareForRecycle` method already sets the `_viewController` to nil which sets the detents to nil automatically. So we're in good shape I think.

Here's the attached video, on how it looks like with states:

https://user-images.githubusercontent.com/47336142/196372310-3a3e9e27-b304-4608-b711-9c9ee20b221e.mp4